### ISSUE TITLE
Incompatible ipset protocols

### ISSUE THREAD

**Issue by joskuijpers** [projectcalico/calico#5717]
When using openSuse MicroOS Kubic calico fails to use upsets:

```
2022-03-06 16:26:02.356 [ERROR][20604] felix/ipsets.go 574: Bad return code from ‘ipset list’. error=exit status 1 family=“inet” stderr=“ipset v7.1: Kernel and userspace incompatible: settype hash:ip with revision 5 not supported by userspace.\n”
2022-03-06 16:26:02.356 [WARNING][20604] felix/ipsets.go 322: Failed to resync with dataplane error=exit status 1 family=“inet”
```

## Expected Behavior
Calico works and has no errors in its logs.

## Possible Solution
-

## Steps to Reproduce (for bugs)
1. Install openSuse Kubic
2.Install a cluster
3.Install Calico (I used the 'less than 50 on bare metal'

## Context
I am trying to build a cluster on Hetzner using minimal OS. Kubic comes with Kubernetes pre-installed so that saves on some installation steps.

## Your Environment
* Calico version: 3.22.1
* Orchestrator version (e.g. kubernetes, mesos, rkt): k8s 1.23
* Operating System and version: openSUSE MicroOS Kubic latest
* Link to your project (optional):

**Comment by lmm** [comment]
It looks like you're hitting https://github.com/projectcalico/calico/issues/5011 which according to our release notes was fixed in Calico v3.22.1
https://projectcalico.docs.tigera.io/release-notes/#v3221

**Comment by lwr20** [comment]
Except that OP states that he's using 
> Calico version: 3.22.1

**Comment by lmm** [comment]
@lwr20 yeah it looks like there are more reports of more reports of some ipset incompatibility in v3.22.1 in #5011

**Comment by joskuijpers** [comment]
Is there anything I can do to help here? I can provide root access to my temporary servers that have this issue if needed.

I am now considering using Flannel, and otherwise looking for another lightweight host OS.

**Comment by fasaxc** [comment]
@joskuijpers Are you sure you're on v3.22.1.  The error message above says `ipset v7.1` but v3.22.1 shipped with `ipset v7.11`

What is the version of `ipset` on the host itself, `ipset version` should show it.

**Comment by xmudrii** [comment]
We've tested it and it's fixed on amd64 clusters, however, this is still an issue for non-amd64 clusters (e.g. arm64). This is as expected because the original PR fixed it only for amd64: https://github.com/projectcalico/calico/pull/5485#discussion_r789876954

Is there a chance to get this fix for other architectures as well?

@joskuijpers Is your cluster a non-amd64 cluster by any chance?

**Comment by frozenprocess** [comment]
Oh boy, more `arm64` compilation steps.
@caseydavenport I'm afraid this will push the `arm64` compilation time even higher.

**Comment by caseydavenport** [comment]
> @caseydavenport I'm afraid this will push the arm64 compilation time even higher.

Such is life. :sob:

**Comment by ScOut3R** [comment]
This was fixed for `arm64` in v3.23.2. The upcoming v3.22.4 will also have the fix when released.
